purchase_receipt_progress/models/purchase_order.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderReceivedByCategory(models.Model):
+    _name = "purchase.order.received.by.category"
+    _description = "Purchase Order Received by Category"
+
+    order_id = fields.Many2one("purchase.order")
+    category_id = fields.Many2one("product.category")
+    product_qty = fields.Float()
+    qty_received = fields.Float()
+    to_receive = fields.Float()
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    purchased_quantity_total = fields.Float(
+        string="Total Quantity", compute="_compute_purchased_quantity_total"
+    )
+    delivery_status = fields.Float(compute="_compute_delivery_status")
+    received_by_category_ids = fields.One2many(
+        "purchase.order.received.by.category",
+        "order_id",
+        string="Received by Category",
+        compute="_compute_received_by_category_ids",
+    )
+
+    @api.depends("order_line.product_qty")
+    def _compute_purchased_quantity_total(self):
+        for order in self:
+            order.purchased_quantity_total = sum(order.mapped("order_line.product_qty"))
+
+    @api.depends("order_line.product_qty", "order_line.qty_received")
+    def _compute_delivery_status(self):
+        for order in self:
+            received_quantity = sum(order.mapped("order_line.qty_received"))
```

Comment:
consider using read_group outside of the loop for performance reason.

Reply:
I didn't get it. Can you show me a example ?